monai/apps/mmars.py

```diff
@@ -0,0 +1,154 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import warnings
+from typing import Mapping
+
+import torch
+
+import monai.networks.nets as monai_nets
+from monai.apps.utils import download_and_extract
+
+__all__ = ["download_mmar", "load_from_mmar", "MODEL_DESC"]
+
+
+MODEL_DESC = (
+    {
+        "id": "clara_pt_prostate_mri_segmentation_1",
+        "name": "clara_pt_prostate_mri_segmentation",
+        "url": "https://api.ngc.nvidia.com/v2/models/nvidia/med/clara_pt_prostate_mri_segmentation/versions/1/zip",
+        "doc": "https://ngc.nvidia.com/catalog/models/nvidia:med:clara_pt_prostate_mri_segmentation",
+        "file_type": "zip",
+        "hash_type": "md5",
+        "hash_val": None,
+        "model_file": os.path.join("models", "model.pt"),
+    },
+    {
+        "id": "clara_pt_covid19_ct_lesion_segmentation_1",
+        "name": "clara_pt_covid19_ct_lesion_segmentation",
+        "url": "https://api.ngc.nvidia.com/v2/models/nvidia/med/clara_pt_covid19_ct_lesion_segmentation/versions/1/zip",
+        "doc": "https://ngc.nvidia.com/catalog/models/nvidia:med:clara_pt_covid19_ct_lesion_segmentation",
+        "file_type": "zip",
+        "hash_type": "md5",
+        "hash_val": None,
+        "model_file": os.path.join("models", "model.pt"),
+    },
+)
+"""
+Collection of the remote MMAR descriptors
+"""
+
+
+def _get_model_spec(idx):
+    """get model specification by `idx`. `idx` could be index of the constant tuple of dict or the actual model ID."""
+    if isinstance(idx, int):
+        return MODEL_DESC[idx]
+    if isinstance(idx, str):
+        key = idx.strip().lower()
+        for cand in MODEL_DESC:
+            if cand["id"].strip().lower() == key:
+                return cand
+    print(f"Available specs are: {MODEL_DESC}.")
+    raise ValueError(f"Unknown MODEL_DESC request: {idx}")
+
+
+def download_mmar(item, mmar_dir=None, progress: bool = True):
+    """
+    Download and extract Medical Model Archive (MMAR) from Nvidia Clara Train.
+
+    See Also:
+        - https://docs.nvidia.com/clara/
+        - Nvidia NGC Registry CLI
+
+    Args:
+        item: the corresponding model item from `MODEL_DESC`.
+        mmar_dir: target directory to store the MMAR, default is mmars subfolder under `torch.hub get_dir()`.
+        progress: whether to display a progress bar.
+
+    Examples::
+        >>> from monai.apps import download_mmar
+        >>> download_mmar("clara_pt_prostate_mri_segmentation_1", mmar_dir=".")
+
+    Returns:
+        The local directory of the downloaded model.
+    """
+    if not isinstance(item, Mapping):
+        item = _get_model_spec(item)
+    if not mmar_dir:
+        mmar_dir = os.path.join(torch.hub.get_dir(), "mmars")
```

Comment:
May I know why use `torch.hub` here?

Reply:
yes, pytorch uses that interface for the pretrained models, it's also currently used by all of our models